docs/tutorials/todo-manager/README.mdx

```diff
@@ -0,0 +1,1097 @@
+   - Go to "Roles"
+   - Create an "Admin" role and assign all scopes (`create:todos`, `read:todos`, `delete:todos`)
+   - Create a "User" role and assign only the `create:todos` scope
+
+4. Assign permissions:
+   - Go to "Users"
+   - Select a user
+   - You can either:
+     - Assign roles in the "Roles" tab (recommended)
+     - Or directly assign scopes in the "Permissions" tab
+
+The scopes will be included in the JWT access token's `scope` claim as a space-separated string.
+
+</TabItem>
+<TabItem value="oauth-oidc" label="OAuth 2.0 / OIDC">
+
+OAuth 2.0 / OIDC providers typically support scope-based access control. When implementing RBAC:
+
+1. Define your required scopes in your authorization server
+2. Configure your client to request these scopes during the authorization flow
+3. Ensure your authorization server includes the granted scopes in the access token
+4. The scopes are usually included in the JWT access token's `scope` claim
+
+Check your provider's documentation for specific details on:
+- How to define and manage scopes
+- How scopes are included in the access token
+- Any additional RBAC features like role management
+
+</TabItem>
+</Tabs>
+
+### Validating tokens and checking permissions
+
+When your MCP server receives a request, it needs to:
+
+1. Validate the access token's signature and expiration
+2. Extract the scopes from the validated token
+3. Check if the token has the required scopes for the requested operation
+
+For example, if a user wants to create a new todo item, their access token must include the `create:todos` scope. Here's how the flow works:
+
+```mermaid
+sequenceDiagram
+    participant Client
+    participant MCP Server
+    participant Auth Server
+
+    Client->>MCP Server: Request with access token
+    
+    alt JWT Validation
+        MCP Server->>Auth Server: Fetch JWKS
+        Auth Server-->>MCP Server: Return JWKS
+        MCP Server->>MCP Server: Validate JWT locally
+    else Token Introspection
+        MCP Server->>Auth Server: POST /introspect<br/>(token=access_token)
+        Auth Server-->>MCP Server: Return token info<br/>(active, scope, etc.)
+    end
+    
+    MCP Server->>MCP Server: Extract & check scopes
+    
+    alt Has required scopes
+        MCP Server->>Client: Allow operation
+    else Missing scopes
+        MCP Server->>Client: Return 403 Forbidden
+    end
+```
+
+### Dynamic Client Registration
+
+Dynamic Client Registration is not required for this tutorial, but it can be useful if you want to automate the MCP client registration process with your authorization server. Check [Is Dynamic Client Registration required?](../../provider-list.mdx#is-dcr-required) for more details.
+
+## Understand RBAC in Todo Manager
+
+For demonstration purposes, we'll implement a simple Role-Based Access Control (RBAC) system in our todo manager MCP server. This will show you the basic principles of RBAC while keeping the implementation straightforward.
```

Comment:
```suggestion
For demonstration purposes, we'll implement a simple role-based access control (RBAC) system in our todo manager MCP server. This will show you the basic principles of RBAC while keeping the implementation straightforward.
```